File: leetcode/java/1470. Shuffle the Array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        // return mySolution(nums, n);
        return leetCodeSolution(nums, n);
    }
    
    private int[] mySolution(int[] nums, int n) {
        final List<int[]> list = new ArrayList<>();
        
        for(int i = 0; i < n; i++) {
            list.add(new int[]{nums[i], nums[i + n]});
        }
        
        final int[] ret = new int[nums.length];
        
        int index = 0;
        for(int[] array : list) {
            ret[index++] = array[0];
            ret[index++] = array[1];
        }
        
        return ret;
    }
    
    private int[] leetCodeSolution(int[] nums, int n) {
        int[] ret = new int[nums.length];
        
        int index = 0;
        int retIndex = 0;
        while(retIndex < nums.length) {
            ret[retIndex++] = nums[index++];
            ret[retIndex++] = nums[n++];
        }
        
        return ret;
    }
}

File: leetcode/kotlin/1470. Shuffle the Array.kt
class Solution {
    fun shuffle(nums: IntArray, n: Int): IntArray {
        // return mySolution1(nums, n)
        return leetCodeSolution(nums, n)
    }
    
    fun mySolution1(nums: IntArray, n: Int): IntArray {
                
        val list = mutableListOf<Int>()
        for (index in 0 until n) {
            list.add(nums.get(index))
            list.add(nums.get(index+n))
        }
        
        val ret = IntArray(nums.count())
        var index = 0
        list.forEach {
            ret.set(index++, it)
        }
        
        return ret
    }
    
    fun leetCodeSolution(nums: IntArray, n: Int): IntArray {
        
        val ret = IntArray(nums.count())
        var retIndex = 0
        var index = 0
        var nIndex = n
        while (index < n) {
            ret[retIndex++] = nums.get(index++)
            ret[retIndex++] = nums.get(nIndex++)
        }
        
        return ret
    }
}
